src/dataset.h

```diff
@@ -488,6 +488,8 @@ class DataSet : public PointArray, public QObject {
 
 };
 
+QDataStream& operator<<(QDataStream& out, const DataSet& dataset);
+QDataStream& operator>>(QDataStream& in, DataSet& dataset);
```

Comment:
My understanding of this warning (#105) is that we define this function as a friend for the class, but we don't actually provide a forward declaration of the function. With recent g++'s, you need to do both. You can't implicitly forward-declare the function by adding it as a friend function to a class.